pkg/apis/migration/v1alpha1/migcluster_types.go

```diff
@@ -77,16 +76,18 @@ func (m *MigCluster) GetServiceAccountSecret(client k8sclient.Client) (*kapi.Sec
 
 // GetClient get a local or remote client using a MigCluster and an existing client
 func (m *MigCluster) GetClient(c k8sclient.Client) (k8sclient.Client, error) {
+	// TODO: re-enable cache after issue with caching secrets is resolved
+
 	// If MigCluster isHostCluster, reuse client
-	if m.Spec.IsHostCluster {
-		return c, nil
-	}
+	// if m.Spec.IsHostCluster {
+	// 	return c, nil
+	// }
```

Comment:
Disabling this part failed for me, since this causes `BuildRestConfig` to fail on the nil `Spec.ClusterRef`